src/Datadog.Trace.ClrProfiler.Managed/Emit/ObjectExtensions.cs

```diff
@@ -30,14 +30,14 @@ internal static class ObjectExtensions
             var paramType1 = typeof(TArg1);
 
             object cachedItem = Cache.GetOrAdd(
-                $"{type.AssemblyQualifiedName}.{methodName}.{paramType1.AssemblyQualifiedName}",
```

Comment:
I'm assuming that the comparer for System.Type has the same granularity as the assembly qualified name. It would make sense, but I need to double-check

Reply:
The AssemblyQualified type was also useful because we want the owning Assembly as well. This is something that we can run into with the application using both `StackExchange.Redis.StrongName` and `StackExchange.Redis` NuGet packages

Reply:
I assumed as much yes. That's why I need to make absolutely sure that the Type equality comparer takes that point into account.

Reply:
Looking good. I created a library and compiled it twice: once in debug without signature, once in release with a strong name (the debug/release distinction is probably irrelevant)

Then ran the following code:

```csharp
var debug = Assembly.LoadFrom(@"C:\Users\kevin.gosse\source\repos\ConsoleApp1\Library1\bin\Debug\Library1.dll");
var release = Assembly.LoadFrom(@"C:\Users\kevin.gosse\source\repos\ConsoleApp1\Library1\bin\Release\Library1.dll");

var typeDebug = debug.GetType("Library1.Class1");
var typeRelease = release.GetType("Library1.Class1");

Console.WriteLine(debug.FullName);
Console.WriteLine(release.FullName);

Console.WriteLine($"{typeDebug.FullName} - {typeDebug.AssemblyQualifiedName}");
Console.WriteLine($"{typeRelease.FullName} - {typeRelease.AssemblyQualifiedName}");

Console.WriteLine(typeDebug.Equals(typeRelease));
```

The output is:

```
Library1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
Library1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=c28eafe45d72d6c9
Library1.Class1 - Library1.Class1, Library1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
Library1.Class1 - Library1.Class1, Library1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=c28eafe45d72d6c9
False
```

So it looks like the type equality comparer does the necessary checks.